hibernate-reactive-core/src/main/java/org/hibernate/reactive/ReactiveLoaderBasedLoader.java

```diff
@@ -0,0 +1,56 @@
+/* Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ * Copyright: Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.reactive;
+
+import org.hibernate.engine.spi.EntityKey;
+import org.hibernate.engine.spi.QueryParameters;
+import org.hibernate.engine.spi.SessionFactoryImplementor;
+import org.hibernate.engine.spi.SharedSessionContractImplementor;
+import org.hibernate.persister.collection.CollectionPersister;
+import org.hibernate.persister.entity.EntityPersister;
+import org.hibernate.reactive.loader.ReactiveLoader;
+import org.hibernate.transform.ResultTransformer;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+
+/**
+ * An interface intended for {@link ReactiveLoader} implementations that also
+ * extend a {@link org.hibernate.loader.Loader} implementation.
```

Comment:
@gbadner do we actually have any `ReactiveLoader`s which aren’t `Loader`s?

Reply:
@gavinking , yes, `ReactivePlanEntityLoader` implements `ReactiveLoader`, but does not extend `Loader`.

Reply:
Ah OK of course, I forgot about that one :)